data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java

```diff
@@ -109,24 +109,23 @@ public void handle(@NotNull final Context ctx) throws Exception {
   public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     final SafeFuture<Optional<ObjectAndMetaData<Fork>>> future =
         chainDataProvider.getFork(request.getPathParameter(PARAMETER_STATE_ID));
-    request.handleOptionalResult(future, this::handleResult, SC_NOT_FOUND);
-  }
-
-  private Optional<String> handleResult(Context ctx, final ObjectAndMetaData<Fork> response)
-      throws JsonProcessingException {
-    return Optional.of(
-        JsonUtil.serialize(
-            new StateForkData(response.isExecutionOptimisticForApi(), response.getData()),
-            RESPONSE_TYPE));
+    request.respondAsync(
+        future.thenApplyChecked(
+            maybeFork -> {
+              if (maybeFork.isEmpty()) {
+                return AsyncApiResponse.respondWithError(SC_NOT_FOUND, "Not found");
+              }
+              return AsyncApiResponse.respondOk(new StateForkData(maybeFork.get()));
+            }));
```

Comment:
nit: Not sure if the indentation will ruin this but could do:

```suggestion
            maybeFork -> maybeFork.map(fork -> new AsyncApiResponse.respondOk(new StateForkData(fork))).orElseGet(() -> AsyncApiResponse.respondWithError(SC_NOT_FOUND, "Not found")));
```
(no chance that passes spotless as is)

And it may be worth having a `AsyncApiResponse.notFound()` method to simplify as well.

Reply:
yep - that'll get changed next PR... will leave for now...